### DISCUSSION TITLE
METIS vs R-tree

### DISCUSSION THREAD

**Discussion by fdrmrc** [fdrmrc/non_matching_test_suite#40] [Show and tell]
@luca-heltai  I set up some test cases meant to compare METIS and R-tree based partitioning for what concerns L2 rates and quality of the solution.

- - - - - - - - - - - - -

**EXAMPLE: STRUCTURED SQUARE partitioned with both METIS or R-tree**

- $\Omega = [0,1]^2$
- $u=xy(x-1)(y-1)$

![immagine](https://github.com/fdrmrc/AggloDeal/assets/50499769/7298fc53-3063-445a-819e-8284a802464b)

- - - - - - - - - - - - -

**EXAMPLE: _UNSTRUCTURED_ SQUARE (gmsh generated) partitioned with both METIS or R-tree**

- $\Omega = [0,1]^2$
- $u=xy(x-1)(y-1)$

![immagine](https://github.com/fdrmrc/AggloDeal/assets/50499769/b87bac8b-7c27-4c45-a540-b603d1f3c661)

- - - - - - - - - - - - -
**EXAMPLE 1: STRUCTURED BALL (radial solution)**

- $\Omega = B_1(0,0)$
- $u=x^2+y^2-1$

![immagine](https://github.com/fdrmrc/AggloDeal/assets/50499769/8c1f0f80-b696-4f4a-a680-b636324b1230)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

I believe the bad behaviour for METIS may be due to the quality of the elements. Here's a zoom of the finest grid obtained with METIS with the structured square (2048 agglomerates):
![immagine](https://github.com/fdrmrc/AggloDeal/assets/50499769/37b9b62c-f764-4744-8caf-058a7bbcd174)

**Comment by luca-heltai** [comment]
These are fantastic news. Did you show them to Andrea as well? Can you start preparing an overleaf with these?

**Reply by luca-heltai** [reply]
Btw, in the structured ball, I only see "L2 error". Is this rtree or metis?

**Reply by fdrmrc** [reply]
It's R-tree. I used METIS only with squares in order to highlight the issue.

**Reply by fdrmrc** [reply]
I updated the `structured ball` with the METIS version. The behaviour looks the same as before

**Comment by fdrmrc** [comment]
> These are fantastic news. Did you show them to Andrea as well? Can you start preparing an overleaf with these?

Yes, I've shown him one hour ago. I believe that the issue with METIS is that our _fine mesh_ is not fine enough. If you look at page 24 of this paper (https://www.esaim-m2an.org/articles/m2an/abs/2016/03/m2an150070/m2an150070.html) by Andrea and collaborators, they start in 3D from a fine mesh of 1 019 674 tets, and use METIS to generate a sequence of (quoting from the paper) 
> 64, 512, 4096, and 32768 elements

Indeed, we observe the right convergence rate for our 2D tests when the number of polygons is "much lower" compared to the size of the true underlying mesh
